### PR TITLE
Cherry-pick #16348 to 7.x: Fix loading processors from autodiscover hints

### LINKED ISSUE #16343: [Filebeat] Autodiscovery logs.processors does not convert string into a MapStr
When using labels on Docker or annotations on Kubernetes to add processors to a container the runner fails to be created because the value of the annotation name is not converted from a `string` to a `common.MapStr` before passing the information to the runner.

Use the follow configuration:

```
filebeat.autodiscover:
  providers:
    - type: docker
      hints.enabled: true

output.console:
  pretty: true
```

Start the container:

`docker run -l 'co.elastic.logs/processors.add_fields={"fields":{"foo":"bar"}}' busybox echo hello`

Errors with:

`Error creating runner from config: can not convert 'string' into 'object' accessing 'processors.0.add_fields'`

### PR DESCRIPTION
**Pull request by blakerouse**
Cherry-pick of PR #16348 to 7.x branch. Original message: 

## What does this PR do?

Fixes an issue where processors from autodiscover hints where not convered into a `common.MapStr`.

## Why is it important?

Allows processors to be defined in autodiscover hints.

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- [X] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist

- [X] Converts `string` to `common.MapStr`

## How to test this PR locally

Run filebeat with:

```
filebeat.autodiscover:
  providers:
    - type: docker
      hints.enabled: true

output.console:
  pretty: true
```

Start container with processor:

`docker run -l 'co.elastic.logs/processors.add_fields={"fields":{"foo":"bar"}}' busybox echo hello`

Check that filebeat was able to start logging it the container with the fields added.

## Related issues

- Closes #16343

### CODE REVIEW

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -75,6 +75,8 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - Remove superfluous use of number_of_routing_shards setting from the default template. {pull}16038[16038]
 - Fix index names for indexing not always guaranteed to be lower case. {pull}16081[16081]
 - Upgrade go-ucfg to latest v0.8.1. {pull}15937{15937}
+- Add `ssl.ca_sha256` option to the supported TLS option, this allow to check that a specific certificate is used as part of the verified chain. {issue}15717[15717]
```
Comment (kaiyan-sheng) [inline CHANGELOG.next.asciidoc]: This changelog entry doesn't belong to this PR 😄

Comment (blakerouse) [reply CHANGELOG.next.asciidoc]: Dang it! Fixed!
